opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -519,31 +522,33 @@ namespace Opm
     template <typename TypeTag>
     void
     MultisegmentWell<TypeTag>::
-    initSegmentRatesWithWellRates(WellState& well_state) const
+    scaleSegmentRatesAndPressureWithWellRatesAndPressure(WellState& well_state) const
```

Comment:
It is not clear to me that this function is a direct replacement. The replaced initSegmentRatesWithWellRates() assumed useful `wellRates()`, and tried to set sensible `perfPhaseRates()` and `segRates()`. The new function assumes useful `wellRates()` and `bhp()`, and sets `segRates()` and `segPress()`. So `perfPhaseRates()` is not touched, is this intentional? Perhaps it is not used and does not matter? Furthermore the `segRates()` are unchanged if the top one is close to zero. I assume that is intentional to account for crossflow? But what if the well has been closed, and then gets a positive ORAT, would it not be better to set the `segRates()` using the previous code rather than doing nothing?

Reply:
> So perfPhaseRates() is not touched, is this intentional?
Yes. perfPhaseRates() is for output only. 

> But what if the well has been closed, and then gets a positive ORAT, would it not be better to set the segRates() using the previous code rather than doing nothing

I guess so. At-least it should not hurt. I will put back the old code and use it when segRates in topsegment < eps. 
I have some plans with refactoring solveWellEq and use it only for initializing new and resurrected wells. Than we may remove the initSegmentRateWithWellRates(...) code and simplify the initializing happening within the wellstate.